netmaster/master/netmaster.go

```diff
@@ -501,17 +498,34 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 		// XXX: do configuration check, to make sure it is allowed
 	}
 
+	subnetIsAllocated := false
 	if nwCfg.SubnetIP == "" {
 		nwCfg.SubnetLen = gCfg.Auto.AllocSubnetLen
 		nwCfg.SubnetIP, err = gCfg.AllocSubnet(rm)
 		if err != nil {
 			return err
 		}
+		subnetIsAllocated = true
 	}
 
-	nwCfg.DefaultGw = network.DefaultGw
-	if nwCfg.DefaultGw == "" {
-		// TBD: allocate per global policy
+	defaultNwName, err := gCfg.AssignDefaultNetwork(network.Name)
+	if err != nil {
+		log.Errorf("error assigning the default network. Error: %s", err)
+		return err
+	}
+
+	if network.Name == defaultNwName {
+		nwCfg.DefaultGw = network.DefaultGw
+		// For auto derived subnets assign gateway ip be the last valid unicast ip the subnet
+		if nwCfg.DefaultGw == "" && subnetIsAllocated {
```

Comment:
just wondering, if this check is really required here.

Would `subnetIsAllocated` be false when there is a user-defined gateway provided? If so, I think in either case can treat it as a default gateway, right?